retrievalmarket/impl/clientstates/client_fsm.go

```diff
@@ -209,8 +209,8 @@ var ClientEvents = fsm.Events{
 		FromMany(rm.DealStatusSendFunds, rm.DealStatusSendFundsLastPayment).To(rm.DealStatusOngoing).
```

Comment:
Need to finalize and cleanup the blockstore even if the deal fails for whatever reason after entering the fsm.